cpp/include/raft/linalg/eig.cuh

```diff
@@ -112,6 +114,117 @@ void eigJacobi(const raft::handle_t& handle,
+ *        for the column-major symmetric matrices
+ * @tparam ElementType the data-type of input and output
+ * @tparam IntegerType Integer used for addressing
+ * @param handle raft::handle_t
+ * @param in input raft::device_matrix_view (symmetric matrix that has real eig values and
+ * vectors)
+ * @param n_eig_vals: number of eigenvectors to be generated
+ * @param eig_vectors: eigenvectors output of type raft::device_matrix_view
+ * @param eig_vals: eigen values output of type raft::device_vector_view
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+void eig_dc_select(const raft::handle_t& handle,
+                   const raft::device_matrix_view<ElementType, IndexType, raft::col_major> in,
+                   std::size_t n_eig_vals,
+                   raft::device_matrix_view<ElementType, IndexType, raft::col_major> eig_vectors,
+                   raft::device_vector_view<ElementType, IndexType> eig_vals,
+                   EigVecMemUsage memUsage)
+{
+  RAFT_EXPECTS(in.is_exhaustive(), "Input must be contiguous");
+  RAFT_EXPECTS(eig_vectors.is_exhaustive(), "Eigen Vectors must be contiguous");
+  RAFT_EXPECTS(eig_vals.is_exhaustive(), "Eigen Values must be contiguous");
+  RAFT_EXPECTS(eig_vectors.size() == n_eig_vals * in.extent(0),
+               "Size mismatch between Input and Eigen Vectors");
+  RAFT_EXPECTS(eig_vals.size() == n_eig_vals, "Size mismatch between Input and Eigen Values");
+
+  raft::linalg::eigSelDC(handle,
+                         in.data_handle(),
+                         in.extent(0),
+                         in.extent(1),
+                         n_eig_vals,
+                         eig_vectors.data_handle(),
+                         eig_vals.data_handle(),
+                         memUsage,
+                         handle.get_stream());
+}
+
+/**
+ * @brief overloaded function for eig decomp with Jacobi method for the
+ * column-major symmetric matrices (in parameter)
+ * @tparam ElementType the data-type of input and output
+ * @tparam IntegerType Integer used for addressing
+ * @param handle raft::handle_t
+ * @param in input raft::device_matrix_view (symmetric matrix that has real eig values and
+ * vectors)
+ * @param eig_vectors: eigenvectors output of type raft::device_matrix_view
+ * @param eig_vals: eigen values output of type raft::device_vector_view
+ * @param tol: error tolerance for the jacobi method. Algorithm stops when the
+ * error is below tol
+ * @param sweeps: number of sweeps in the Jacobi algorithm. The more the better
+ * accuracy.
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+void eig_jacobi(const raft::handle_t& handle,
+                const raft::device_matrix_view<ElementType, IndexType, raft::col_major> in,
+                raft::device_matrix_view<ElementType, IndexType, raft::col_major> eig_vectors,
+                raft::device_vector_view<ElementType, IndexType> eig_vals,
+                ElementType tol = 1.e-7,
```

Comment:
1. Would you consider passing multiple algorithm parameters as a struct?  Users could then use designated initializers to name the input values, if they want to specify any nondefault values.

2. Would you consider making the default value of `tol` depend on `ElementType`, by using `numeric_limits::epsilon()`?

```c++
template<class ValueType>
struct eig_jacobi_params {
  using magnitude_type = decltype(std::abs(std::declval<ValueType>()));

  magnitude_type tol = std::numeric_limits<magnitude_type>::epsilon();
  int sweeps = 15;
};
```